src/coreclr/jit/block.h

```diff
@@ -1291,6 +1254,47 @@ struct BasicBlockList
     }
 };
 
+// BBswtDesc -- descriptor for a switch block
+//
+//  Things to know:
+//  1. If bbsHasDefault is true, the default case is the last one in the array of basic block addresses
+//     namely bbsDstTab[bbsCount - 1].
+//  2. bbsCount must be at least 1, for the default case. bbsCount cannot be zero. It appears that the ECMA spec
+//     allows for a degenerate switch with zero cases. Normally, the optimizer will optimize degenerate
+//     switches with just a default case to a BBJ_ALWAYS branch, and a switch with just two cases to a BBJ_COND.
+//     However, in debuggable code, we might not do that, so bbsCount might be 1.
+//
+struct BBswtDesc
+{
+    BasicBlock**         bbsDstTab; // case label table address
+    unsigned             bbsCount;  // count of cases (includes 'default' if bbsHasDefault)
+    unsigned             bbsDominantCase;
+    BasicBlock::weight_t bbsDominantFraction;
+    bool                 bbsHasDefault;
+    bool                 bbsHasDominantCase;
+
+    BBswtDesc() : bbsHasDefault(true), bbsHasDominantCase(false)
+    {
+    }
```

Comment:
You need to update `optCopyBlkDest` for the new fields (probably?) -- looks like that code is already broken for `bbsHasDefault`. `BBswtDesc` should probably have a copy function.

Reply:
Should you dump the new fields in `dspJumpKind`? `fgTableDispBasicBlock`?

Reply:
Fixed and added some output to both methods, eg
```
BB01 [0000]  1                            56     56    [000..022)-> BB03,BB05,BB04[dom(0.5535714)],BB06,BB08,BB07,BB09,BB02[def] (switch)                     IBC 
```